diff --git a/Makefile b/Makefile
index c3ba9be..14e3001 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,11 @@
-obj-m += acer-wmi-battery.o
-PWD := $(CURDIR)
+KERNELRELEASE ?= $(shell uname -r)
+KERNEL_DIR  ?= /lib/modules/$(KERNELRELEASE)/build
+PWD := $(shell pwd)
+
+obj-m := acer-wmi-battery.o
 
 all:
-	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
+	$(MAKE) -C $(KERNEL_DIR) M=$(PWD) modules
 
-clean:
-	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
+install:
+	$(MAKE) -C $(KERNEL_DIR) M=$(PWD) modules_install
